rfcs/2020-08-31-3641-mongo-metrics.md

```diff
@@ -0,0 +1,801 @@
+- `mongodb_mongod_wiredtiger_cache_max_bytes` (gauge)
+- `mongodb_mongod_wiredtiger_cache_overhead_percent` (gauge)
+- `mongodb_mongod_wiredtiger_cache_pages` tagged with type (gauge)
+- `mongodb_mongod_wiredtiger_cache_pages_total` tagged with type (counter)
+- `mongodb_mongod_wiredtiger_concurrent_transactions_available_tickets` tagged with type (gauge)
+- `mongodb_mongod_wiredtiger_concurrent_transactions_out_tickets` tagged with type (gauge)
+- `mongodb_mongod_wiredtiger_concurrent_transactions_total_tickets` tagged with type (gauge)
+- `mongodb_mongod_wiredtiger_log_bytes_total` tagged with type (counter)
+- `mongodb_mongod_wiredtiger_log_operations_total` tagged with type (counter)
+- `mongodb_mongod_wiredtiger_log_records_scanned_total` (counter)
+- `mongodb_mongod_wiredtiger_log_records_total` tagged with type (counter)
+- `mongodb_mongod_wiredtiger_session_open_cursors` (gauge)
+- `mongodb_mongod_wiredtiger_session_open_sessions` (gauge)
+- `mongodb_mongod_wiredtiger_transactions_checkpoint_seconds` tagged with type (gauge)
+- `mongodb_mongod_wiredtiger_transactions_checkpoint_seconds_total` (counter)
+- `mongodb_mongod_wiredtiger_transactions_running_checkpoints` (gauge)
+- `mongodb_mongod_wiredtiger_transactions_total` tagged with type (counter)
+- `mongodb_network_bytes_total` tagged with state (counter)
+- `mongodb_network_metrics_num_requests_total` (counter)
+- `mongodb_op_counters_repl_total` tagged with type (counter)
+- `mongodb_op_counters_total` tagged with type (counter)
+
+All metrics will also be tagged with the `endpoint` (stripped of user name & password if authentication is used) and `host`.
+
+## Doc-level Proposal
+
+The following additional source configuration will be added:
+
+```toml
+[sources.my_source_id]
+  type = "mongodb_metrics" # required
+  endpoint = "mongodb://mongodb_exporter:s3cr3tpassw0rd@localhost:27017" # required - address of the MongoDBserver.
+  scrape_interval_secs = 15 # optional, default, seconds
+  namespace = "mongodb" # optional, default is "mongodb", namespace to attach to metrics.
+```
+
+We will also expose the TLS settings and support `tls` in the `endpoint` URL.
+
+- We'd also add a guide for doing this with MongoDB Authorization.
+
+## Rationale
+
+MongoDB is a commonly adopted modern database. Users frequently want to monitor its state and performance.
+
+Additionally, as part of Vector's vision to be the "one tool" for ingesting and shipping observability data, it makes sense to add as many sources as possible to reduce the likelihood that a user will not be able to ingest metrics from their tools.
+
+## Prior Art
+
+- https://github.com/dcu/mongodb_exporter
+- https://github.com/percona/mongodb_exporter
+- https://github.com/influxdata/telegraf/tree/master/plugins/inputs/mongodb
+- https://collectd.org/wiki/index.php/Plugin:MongoDB
+- https://docs.datadoghq.com/integrations/mongo/?tab=host
+
+### mongodb_exporter output
```

Comment:
Am I missing something, or are these go and exporter specific metrics? I wanted to compare our output with the official Prometheus [mongodb_exporter](https://github.com/dcu/mongodb_exporter). But they don't document the metrics.

Reply:
This is a dump of all the metrics the exporter emits, same for Telegraf. The metrics above are what we get from the various MongoDB server commands.

Reply:
There doesn't seem to include any MongoDB specific metrics though. I just see `go_*` and `mongodb_exporter_*` which appear to describe the Golang environment and the actual exporter activity respectively.

Reply:
Oh weird. They must have been truncated. Re-added.